sympy/functions/elementary/tests/test_hyperbolic.py

```diff
@@ -558,6 +558,10 @@ def test_asinh():
 def test_asinh_rewrite():
     x = Symbol('x')
     assert asinh(x).rewrite(log) == log(x + sqrt(x**2 + 1))
+    assert asinh(x).rewrite(atanh) == atanh(x/sqrt(1 + x**2))
+    assert asinh(x).rewrite(asin) == asinh(x)
+    assert asinh(x*(1 + I)).rewrite(asin) == -I*asin(I*x*(1+I))
+    assert asinh(x).rewrite(acos) == I*(-I*asinh(x) + pi/2) - I*pi/2
```

Comment:
Isn't this odd: no `acos`, just a rewriting in terms of original function?

Reply:
And when expanded, it is just the original.

Reply:
The problem is automatic evaluation:
```python
In [2]: acos(I * x)
Out[2]: 
              π
-ⅈ⋅asinh(x) + ─
              2
```